gt-elasticsearch/src/main/java/mil/nga/giat/data/elasticsearch/ElasticAttribute.java

```diff
@@ -198,8 +223,53 @@ public boolean equals(Object obj) {
             equal &= Objects.equals(analyzed, other.analyzed);
             equal &= Objects.equals(stored, other.stored);
             equal &= Objects.equals(nested, other.nested);
+            equal &= Objects.equals(order, other.order);
+            equal &= Objects.equals(customName, other.customName);
         }
         return equal;
     }
-
+    
+    /**
+     * Implement comparison logic
+     * @param o is a non-null ElasticAttribute
+     * @return negative for before, zero for same, positive after 
+     */
+    @Override
+    public int compareTo(ElasticAttribute o) {
+        if(this.order == null) { 
+            if(o.order == null)
+                return this.name.compareTo(o.name);
+            return 1;
+        }
+        if(o.order == null)
+            return -1;
+     
+        int i = this.order.compareTo(o.order);
+        return i == 0 ? this.name.compareTo(o.name) : i;
+    }  
+
+    /**
+     * Perform basic update to the given name to make it XML namespace
+     * compliant.
+     * 
+     * @param name
+     * @return Name that is XML safe
+     */
+    private static String normalizeName (String name) {
+        
+        String normalName = name;
+        
+        /* XML element naming rules:
+         * 1. Element names must start with a letter or underscore
+         * 2. Element names cannot start with the letters xml
+         * 3. Element names cannot contain spaces
+         */
+        if (normalName.toLowerCase().startsWith("xml")) {
+            normalName = "_".concat(normalName);
+        } else if (! beginLetters.matcher(normalName).matches()) {
+            normalName = "_".concat(normalName);
+        }         
+        /* Simply replace all spaces in the name with "_". */
+        return normalName.replaceAll(" ", "_");
+    }
```

Comment:
Can you add tests to cover the custom order and name logic here (see [coverage report](https://coveralls.io/builds/21169760/source?filename=gt-elasticsearch%2Fsrc%2Fmain%2Fjava%2Fmil%2Fnga%2Fgiat%2Fdata%2Felasticsearch%2FElasticAttribute.java#L238)).

Reply:
OK, I'll add a compare test and update others with the custom name.